node/src/chain/metrics.rs

```diff
@@ -0,0 +1,96 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use node_data::Serializable;
+use std::collections::VecDeque;
+use std::io;
+use std::io::{Read, Write};
+use std::ops::Div;
+use std::time::Duration;
+
+const AVG_VALUES_NUM: usize = 5;
+
+/// AverageElapsedTime calculates the average value of last N values added
+#[derive(Debug)]
+pub struct AverageElapsedTime(VecDeque<Duration>);
+impl AverageElapsedTime {
+    pub fn push_back(&mut self, value: Duration) {
+        if self.0.len() == self.0.capacity() {
+            self.0.pop_front();
+        }
+        self.0.push_back(value);
+    }
+
+    pub fn average(&self) -> Option<Duration> {
+        if self.0.is_empty() {
+            return None;
+        }
+
+        let sum: Duration = self.0.iter().sum();
+
+        let mut res = sum.div(self.0.len() as u32);
+        if res.subsec_millis() >= 500 {
```

Comment:
Why this condition? We should roundup all values IMO.

Reply:
> but when setting the timeout, the average should be rounded up to the second.

Reply:
Maybe I'm misinterpreting the code. In my understanding, this is rounding up only if the fractional part is over 500 ms.  So, for instance, 0.4 would not be roundup. 
Is this the case?

What I mean with rounding up is that even 0.1 becomes 1.

Reply:
You mean ceiling the result value. Done.